example/linux/gw-example/main.c

```diff
@@ -101,11 +101,11 @@ static bool MQTT_publish(char * topic, uint8_t * payload, size_t payload_size, b
 static void * publish_thread(void * unused)
 {
     message_to_publish_t * message;
-    pthread_mutex_lock(&m_queue_mutex);
     while (true)
     {
         if (m_queue_empty)
```

Comment:
Could we rename m_queue_empty -> m_pub_queue_empty ? (to avoid confusion with the other m_queue_empty, and keep consistent with m_pub_queue_read, m_pub_queue_write)

Reply:
Sure!